smslibrary/src/test/java/com/eis/smslibrary/SMSPeerTest.java

```diff
@@ -1,142 +1,134 @@
+    public void invalidPeer_isNotCreated() {
+        new SMSPeer(randomInvalidAddress);
     }
 
     @Test
-    public void compareTo_invertedOrder_isOrdered() {
-        Assert.assertTrue(defaultPeer2.compareTo(defaultPeer1) > 0);
+    public void validAddressed_areCompared() {
+        SMSPeer lowerPeer = new SMSPeer(VALID_ADDR);
+        SMSPeer higherPeer = new SMSPeer(HIGHER_VALID_ADDR);
+        assertTrue(higherPeer.compareTo(lowerPeer) > 0);
     }
 
     @Test
-    public void compareTo_samePeer_isZero() {
-        Assert.assertEquals(defaultPeer1.compareTo(defaultPeer1), 0);
+    public void validAddresses_areGet() {
+        for(String validAddress : randomValidAddresses)
+            assertEquals(validAddress, new SMSPeer(validAddress).getAddress());
     }
 
     @Test
-    public void compareTo_copyPeer_isZero() {
-        Assert.assertEquals(defaultPeer1.compareTo(copyOfPeer1), 0);
+    public void validPeers_areEqual() {
+        assertEquals(new SMSPeer(VALID_ADDR), new SMSPeer(VALID_ADDR_WITH_PREFIX));
+        assertEquals(new SMSPeer(VALID_ADDR_WITH_SPACES), new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_SPACES));
+        assertEquals(new SMSPeer(VALID_ADDR_WITH_DASHES), new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_DASHES));
+        assertEquals(new SMSPeer(EMULATOR_ADDR), new SMSPeer(EMULATOR_ADDR_WITH_EXT));
+        assertEquals(new SMSPeer(EMULATOR_ADDR), new SMSPeer(EMULATOR_ADDR_WITH_PREFIX_AND_EXT));
     }
 
     @Test
-    public void getAddress_isSame() {
-        Assert.assertEquals(VALID_TELEPHONE_NUMBER + "1", defaultPeer1.getAddress());
-    }
-
-    @Test
-    public void getAddress_isNotSame() {
-        Assert.assertNotEquals(VALID_TELEPHONE_NUMBER, defaultPeer1.getAddress());
-    }
-
-    @Test
-    public void equals_self_isTrue() {
-        Assert.assertEquals(defaultPeer1, defaultPeer1);
-    }
-
-    @Test
-    public void equals_copy_isTrue() {
-        Assert.assertEquals(defaultPeer1, copyOfPeer1);
-    }
-
-    @Test
-    public void equals_different_isFalse() {
-        Assert.assertNotEquals(defaultPeer1, defaultPeer2);
-    }
-
-    @Test
-    public void equals_null_isFalse() {
-        Assert.assertNotEquals(defaultPeer1, null);
-    }
-
-    @Test
-    public void equals_differentClass_isFalse() {
-        Assert.assertNotEquals(defaultPeer1, new Object());
+    public void validPeersHashes_areEqual() {
+        assertEquals(new SMSPeer(VALID_ADDR).hashCode(), new SMSPeer(VALID_ADDR_WITH_PREFIX).hashCode());
+        assertEquals(new SMSPeer(VALID_ADDR_WITH_SPACES).hashCode(), new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_SPACES).hashCode());
+        assertEquals(new SMSPeer(VALID_ADDR_WITH_DASHES).hashCode(), new SMSPeer(VALID_ADDR_WITH_PREFIX_AND_DASHES).hashCode());
+        assertEquals(new SMSPeer(EMULATOR_ADDR).hashCode(), new SMSPeer(EMULATOR_ADDR_WITH_EXT).hashCode());
+        assertEquals(new SMSPeer(EMULATOR_ADDR).hashCode(), new SMSPeer(EMULATOR_ADDR_WITH_PREFIX_AND_EXT).hashCode());
     }
 }
-
```

Comment:
If you wanna test some more numbers to see if you missed anything you could add a test trying out a bunch of random numbers. There is a method in PhoneNumberUtil from libphonenumber for that

Reply:
That's what I'm actually doing 😉 see the `setup()` function

Reply:
My bad me blind me dumb me tired